### ISSUE TITLE
No CUDA runtime is found / Found no NVIDIA driver on your system.

### ISSUE THREAD

**Issue by bwhitman** [replicate/cog#280]
When running a GPU cog: 

```
$ sudo cog predict r8.im/allenhung1025/looptest@sha256:f5cd715e99046e0513fe2b4034e8f7d8c102525b02f49efb52b05f46fcb9ea83
Starting Docker image r8.im/allenhung1025/looptest@sha256:f5cd715e99046e0513fe2b4034e8f7d8c102525b02f49efb52b05f46fcb9ea83 and running setup()...
No CUDA runtime is found, using CUDA_HOME='/usr/local/cuda'
Traceback (most recent call last):
...
AssertionError: 
Found no NVIDIA driver on your system. Please check that you
have an NVIDIA GPU and installed a driver from
http://www.nvidia.com/Download/index.aspx
ⅹ Failed to get container status: exit status 1
```

But CUDA is fine:
```
$ cd /usr/local/cuda/samples/1_Utilities/deviceQuery
$ sudo make
$ ./deviceQuery
Detected 1 CUDA Capable device(s)

Device 0: "NVIDIA GeForce RTX 2080 Ti"
  CUDA Driver Version / Runtime Version          11.4 / 11.2
  CUDA Capability Major/Minor version number:    7.5
  Total amount of global memory:                 11016 MBytes (11551440896 bytes)
  (68) Multiprocessors, ( 64) CUDA Cores/MP:     4352 CUDA Cores
...
Result = PASS
```

**Comment by andreasjansson** [comment]
Hi @bwhitman! 👋 

Does it work if you run the model with Docker directly?

```
$ docker run -d -p 5000:5000 --gpus all r8.im/allenhung1025/looptest@sha256:f5cd715e99046e0513fe2b4034e8f7d8c102525b02f49efb52b05f46fcb9ea83

$ curl http://localhost:5000/predict -X POST -F seed=-1
```

**Comment by bwhitman** [comment]
Hm, no: 

```$ docker run -d -p 5000:5000 --gpus all r8.im/allenhung1025/looptest@sha256:f5cd715e99046e0513fe2b4034e8f7d8c102525b02f49efb52b05f46fcb9ea83
f189dd7fdffa728fc77f7c8976a3b9e4e4619cb670ed411774689439bd9e302c
docker: Error response from daemon: could not select device driver "" with capabilities: [[gpu]].```

I'll dig in, doesn't seem like a cog issue

**Comment by bwhitman** [comment]
It's certainly my system's issue but I have fixed this by using `[nvidia-docker](https://github.com/NVIDIA/nvidia-docker)` instead of docker. Does cog allow overriding which docker command to call?

**Comment by andreasjansson** [comment]
Cog doesn't support that, at the moment at least. I'm curious though, did you [install the nvidia-docker toolkit](https://stackoverflow.com/a/58432877/135797) and then `docker run --gpus all` still didn't work? We should catch that and give the user guidance on how to work around that.

**Comment by bwhitman** [comment]
I did install the toolkit, and `docker run --gpus all` does work (same as nvidia-docker, but not with `cog`.)

**Comment by andreasjansson** [comment]
Oh I see! That's super strange. What output do you get if you do `cog run --debug ...`?

**Comment by bwhitman** [comment]
```
$ cog predict --debug  r8.im/allenhung1025/looptest@sha256:f5cd715e99046e0513fe2b4034e8f7d8c102525b02f49efb52b05f46fcb9ea83

Starting Docker image r8.im/allenhung1025/looptest@sha256:f5cd715e99046e0513fe2b4034e8f7d8c102525b02f49efb52b05f46fcb9ea83 and running setup()...
$ docker run --rm --shm-size 8G --detach --publish 5081:5000 r8.im/allenhung1025/looptest@sha256:f5cd715e99046e0513fe2b4034e8f7d8c102525b02f49efb52b05f46fcb9ea83
No CUDA runtime is found, using CUDA_HOME='/usr/local/cuda'
...
```

**Comment by andreasjansson** [comment]
It looks like Cog doesn't realize this model needs GPUs. I'll dig in and fix this. Thanks for the report Brian, super helpful!

**Comment by andreasjansson** [comment]
This is fixed now in 0.0.14. If you [install the latest Cog](https://github.com/replicate/cog#install) it should work!

**Comment by zeke** [comment]
@allcontributors please add @bwhitman for bugs 🐞
